src/transformers/pipelines/image_to_text.py

```diff
@@ -45,7 +45,7 @@ def __init__(self, *args, **kwargs):
         )
 
     def _sanitize_parameters(self, **kwargs):
-        return {}, {}, {}
+        return {}, kwargs, {}
```

Comment:
I don't think we should do that.

Other pipeline have suffered from conflicting parameters very fast because both `tokenizer` and `generate` support `max_length` where they have different meaning. And since we don't allow breaking changes it becomes super hard to change a behavior once it's set in a release.

One way to do this would be to:
- Add a generic `generate_kwargs` which we would expect a raw dict of arguments passed directly to generate without looking into it enabling all kind of usage.
- Add a very few select arguments like `max_new_tokens` (`max_length` is being slowly deprecated as it's much harder to manipulate than `max_new_tokens`).
- Then we have to do the fusion between this first class parameter and `generate_kwargs` and handle the situation where both are defined.

Would that make sense ?

I can take care of the modifications if you want.

Reply:
I'm try to fix your issue by save kwargs to `self._generate_kwargs`.

But I don't think, that try to select few arguments from kwargs here is good idea.

Currenly without fusion to let you check first, that current changes good. Than I can merge keys from `model_inputs`. and `generate_kwargs` and throw exception if they intersect.

Reply:
Hi, you can't do that, the pipeline object is supposed to be stateless. Assigning to `self` is discouraged.
Also really we can't just feed all arguments like that as I mentioned, parameter clashing is a real thing, and it's impossible to disentangle once a "pass-all"  parameter has been accepted.